### DISCUSSION TITLE
Mozilla tts output voice still sounds robotic after almost 400k

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#180] [Archived]
<USER>>>> Yilmaz_Ay<br/> <DATE> [April 13, 2020, 7:22am]<br/> <br/> Hi All, *slash* <br/> I trained my audio set which consists of about 27 hours of auidos of 10<br/> seconds length and 16000 Hz sample rate with Tacotron2. It took about 4<br/> and half days to train. At this stage the test audios still sounds a<br/> little bit robotic. And in some of our test audios some words are<br/> missing. In some audios there are repetitions. When I look at the graphs<br/> on the tensorboard pages, the graphs look normal. *slash* <br/> <br/> *slash* <br/> <br/> *slash* <br/> My config values mostly are according to the default values. Could<br/> anyone have a look at my configs and let me know what could be wrong? Is<br/> there any parameters that I can change to remove robotic sound from the<br/> test values and improve the output waves quality? *slash* <br/> My configs are as blow: *slash* <br/> 'model': 'Tacotron2', *slash* <br/> 'run_name': 'stspeech-stft_params', *slash* <br/> 'run_description': 'tacotron2 constant stf parameters', *slash* <br/> 'audio':{ *slash* <br/> 'num_mels': 80, *slash* <br/> 'num_freq': 1025, *slash* <br/> 'sample_rate': 16000, *slash* <br/> 'win_length': 1024, *slash* <br/> 'hop_length': 256, *slash* <br/> 'frame_length_ms': null, *slash* <br/> 'frame_shift_ms': null, *slash* <br/> 'preemphasis': 0.98, *slash* <br/> 'min_level_db': -100, *slash* <br/> 'ref_level_db': 20, *slash* <br/> 'power': 1.5, *slash* <br/> 'griffin_lim_iters': 30, *slash* <br/> 'signal_norm': true, *slash* <br/> 'symmetric_norm': true, *slash* <br/> 'max_norm': 4.0, *slash* <br/> 'clip_norm': true, *slash* <br/> 'mel_fmin': 0.0, *slash* <br/> 'mel_fmax': 8000.0, *slash* <br/> 'do_trim_silence': true, *slash* <br/> 'trim_db': 60 *slash* <br/> }, *slash* <br/> 'characters':{ *slash* <br/> 'pad': ' *slash* _', *slash* <br/> 'eos': ' *slash* ~', *slash* <br/> 'bos': ' *slash* ^', *slash* <br/> 'characters':<br/> *slash* 'ABCDEFGHIJKLMNOPQRSTUVWXYZÇĞİÖŞÜabcdefghijklmnopqrstuvwxyzçğıöşü!'(),-.:;?<br/> ', *slash* <br/> 'punctuations':'!'(),-.:;? *slash* ', *slash* <br/> 'phonemes':'iyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻʘɓǀɗǃʄǂɠǁʛpbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟˈˌːˑʍwɥʜʢʡɕʑɺɧɚ˞ɫ' *slash* <br/> },<br/> <br/> 'distributed':{<br/> 'backend': 'nccl',<br/> 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> },<br/> <br/> 'reinit_layers': *slash* [ *slash* ], *slash* <br/> 'batch_size': 32, *slash* <br/> 'eval_batch_size':16, *slash* <br/> 'r': 7, *slash* <br/> 'gradual_training': *slash* [ *slash* [0, 7, 64 *slash* ], *slash* [1, 5, 64 *slash* ], *slash* [50000, 3, 32 *slash* ],<br/> *slash* [130000, 2, 32 *slash* ], *slash* [290000, 1, 32 *slash* ] *slash* ], *slash* <br/> 'loss_masking': true, *slash* <br/> 'run_eval': true, *slash* <br/> 'test_delay_epochs': 5, *slash* <br/> 'test_sentences_file': 'tr_sentences.txt',<br/> <br/> 'noam_schedule': false,<br/> 'grad_clip': 1.0,<br/> 'epochs': 1000,<br/> 'lr': 0.00001,<br/> 'wd': 0.000001,<br/> 'warmup_steps': 4000,<br/> 'seq_len_norm': false,<br/> <br/> 'memory_size': -1,<br/> 'prenet_type': 'original',<br/> 'prenet_dropout': true,<br/> <br/> 'attention_type': 'original',<br/> 'attention_heads': 4,<br/> 'attention_norm': 'sigmoid',<br/> 'windowing': false,<br/> 'use_forward_attn': false,<br/> 'forward_attn_mask': false,<br/> 'transition_agent': false,<br/> 'location_attn': false,<br/> 'bidirectional_decoder': false,<br/> 'stopnet': true,<br/> 'separate_stopnet': true,<br/> <br/> 'print_step': 5,<br/> 'save_step': 5000,<br/> 'checkpoint': true,<br/> 'tb_model_param_stats': false,<br/> <br/> 'text_cleaner': 'phoneme_cleaners',<br/> 'enable_eos_bos_chars': false,<br/> 'num_loader_workers': 1,<br/> 'num_val_loader_workers': 1,<br/> 'batch_group_size': 0,<br/> 'min_seq_len': 6,<br/> 'max_seq_len': 150,<br/> <br/> 'output_path': 'train_logs/',<br/> <br/> 'phoneme_cache_path': 'mozilla_tr_phonemes_2_1',<br/> 'use_phonemes': true,<br/> 'phoneme_language': 'tr',<br/> <br/> 'use_speaker_embedding': false,<br/> 'style_wav_for_test': null,<br/> 'use_gst': false,<br/> <br/> 'datasets':<br/> [<br/> {<br/> 'name': 'stspeech',<br/> 'path': 'STS-22K/',<br/> 'meta_file_train': 'metadata_train.csv',<br/> 'meta_file_val': 'metadata_test.csv'<br/> }<br/> ]<br/> <br/> }<br/> <br/> Four config parameters are different than the default values: *slash* <br/> 1: the sample rate *slash* <br/> 2: 'griffin_lim_iters' , I reduced it to 30 the default was 60. I did<br/> this to reduce the training time. *slash* <br/> 3: I reduced number of workers to 1, the defaults were 4. I thought that<br/> is something to the with number of GPUS. Since I have just one GPU, I<br/> thought I need to change them as 1. *slash* <br/> 4: min and max seq length parameters. Actually I forgot to change them<br/> according to the my data's lengths. How much effect does it have on the<br/> quality?<br/> <br/> I appreciate any insights or comments or suggestions about what could be<br/> wrong with my training. *slash* <br/> Many many thanks in advance.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/mozilla-tts-output-voice-still-sounds-robotic-after-almost-400k]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [April 13, 2020, 3:43pm]<br/> <br/> > 2: 'griffin_lim_iters' , I reduced it to 30 the default was 60. I did<br/> > this to reduce the training time.<br/> <br/> GriffinLim has nothing to do with the training speed. It's an algorithm<br/> used to synthesize speech.<br/> <br/> > 3: I reduced number of workers to 1, the defaults were 4. I thought<br/> > that is something to the with number of GPUS. Since I have just one<br/> > GPU, I thought I need to change them as 1.<br/> <br/> num_workers are used to load the data into batches during training, it's<br/> using CPU... so you actually might slow down training setting it to 1...<br/> default 4 should be fine.<br/> <br/> > At this stage the test audios still sounds a little bit robotic.<br/> <br/> Default TTS (using GriffinLim as vocoder) will always sound robotic. *slash* <br/> If you want natural sounding speech... neural vocoders are what you are<br/> looking for.<br/> <br/> Mozilla TTS has adapted <https://github.com/erogol/WaveRNN> or<br/> <https://github.com/erogol/ParallelWaveGAN> as such.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> Yilmaz_Ay<br/> <DATE> [April 14, 2020, 6:42am]<br/> <br/> Hi sanjaesc, *slash* <br/> Thanks a lot. That was very helpful.<br/> <br/> [Archived Post]<br/>
